### PR TITLE
Combine generated and original sentences while rephrasing 

### PR DESCRIPTION
**Pull request by tteofili**
Make the algorithm less rely on left-to-right expansion.

### CODE REVIEW

**Comment by tteofili** [comment]
This first set of changes introduce the possibility of using the original piece of sentence to help the model when the `<mask>` token is early in the sentence (giving it more context).

A more straightforward version of bidirectional rephrasing would involve something like the following:
```python
base = self.compute_mask_logits(self.base, original, mask=False)
    logits = base
    for i in range(len(self.experts)):
        logits += self.expert_weights[i] * self.compute_mask_logits(self.experts[i], masked_output, mask=False)
    return self.tokenizer.convert_tokens_to_string(
        self.tokenizer.convert_ids_to_tokens(np.argmax(logits, axis=1)))
```
however this seems to be less effective in practice (lower quality rephrasing), not sure yet why this is the case.

**Comment by tteofili** [comment]
on top of this a new experimental method `rephrase_incremental` has been introduced, it runs the rephrasing incrementally on the original text, first rephrasing, second rephrasing, etc.
